Amplify/Categories/API/AuthProvider/APIAuthProviderFactory.swift

```diff
@@ -18,11 +18,19 @@ open class APIAuthProviderFactory {
     open func oidcAuthProvider() -> AmplifyOIDCAuthProvider? {
         return nil
     }
+
+    open func functionAuthProvider() -> AmplifyFunctionAuthProvider? {
+        return nil
+    }
+}
+
+public protocol AmplifyAuthTokenProvider {
+    typealias AuthToken = String
+    func getLatestAuthToken() -> Result<AuthToken, Error>
```

Comment:
Let's verify that the code path that invokes the token provider wraps any returned error in an Amplify `APIError`. If so, then I'm OK leaving this returning `Error`

Reply:
- `AuthTokenURLRequestInterceptor` (mutations, queries) wraps the returned error in a `APIError.operationError` 
- `OIDCAuthInterceptor` (subscriptions) in case of failure retrieving the token it just skips the auth step and doesn't return/wrap the error